core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,907 @@
+                        .build())
+                .addRepresentations(
+                    ImmutableSQLViewRepresentation.builder()
+                        .sql("select * from ns.tbl using X")
+                        .dialect("trino")
+                        .build())
+                .build());
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void createViewThatAlreadyExists() {
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).isFalse();
+
+    View view =
+        catalog()
+            .buildView(viewIdentifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(viewIdentifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(viewIdentifier)).isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(viewIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .withDefaultNamespace(viewIdentifier.namespace())
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+  }
+
+  @Test
+  public void createViewThatAlreadyExistsAsTable() {
+    TableIdentifier tableIdentifier = TableIdentifier.of("ns", "table");
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).isFalse();
+
+    View view =
+        catalog()
+            .buildView(viewIdentifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(viewIdentifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(tableIdentifier)).isFalse();
+    assertThat(catalog().viewExists(viewIdentifier)).isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(viewIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .withDefaultNamespace(viewIdentifier.namespace())
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+
+    tableCatalog().buildTable(tableIdentifier, SCHEMA).create();
+    assertThat(tableCatalog().tableExists(tableIdentifier)).isTrue();
```

Comment:
Looks like these tests haven't been fixed yet.

The tests should separate different cases:
1. When the catalog supports only views
2. When the catalog also supports tables (that is, `tableCatalog()` is non-null and `supportsTables()` is set)

The second tests can be disabled with `Assume`.

Reply:
For 1) there is `createViewThatAlreadyExists()` and for 2) there is `createViewThatAlreadyExistsAsTable()` and `createTableThatAlreadyExistsAsView()`. However, what was missing is the assumtion to run the tests from 2) only if `tableCatalog()` isn't `null`, which I've added now.